restli-tools/src/main/java/com/linkedin/restli/tools/data/SchemaFormatTranslator.java

```diff
@@ -207,32 +209,31 @@ private void translateFiles() throws IOException, InterruptedException
   private Map<String, SchemaInfo> getTopLevelSchemaToTranslatedSchemaMap() throws IOException
   {
     Map<String, SchemaInfo> topLevelTranslatedSchemas = new HashMap<>();
+    DataSchemaParser dataSchemaParser = new DataSchemaParser(_resolverPath);
+    DataSchemaParser.ParseResult parsedSources = dataSchemaParser.parseSources(
+        new String[]{_sourceDir.getAbsolutePath()});
 
-    Iterator<File> iter = FileUtils.iterateFiles(_sourceDir, new String[]{_sourceFormat}, true);
-    StringBuilder errorMessages = new StringBuilder();
-    while(iter.hasNext())
+    for(Map.Entry<DataSchema, DataSchemaLocation> entry : parsedSources.getSchemaAndLocations().entrySet())
     {
-      File sourceFile = iter.next();
-      String relativePath = _sourceDir.toURI().relativize(sourceFile.toURI()).getPath();
-      String relativeMinusExt = trimFileExtension(relativePath);
-      String schemaFullname = relativeMinusExt.replace(File.separatorChar, '.');
-
-      // When translating files 1:1, a new resolver and parser are required for each file translated
-      // so that a single top level output schema is matched to each input file.
-      MultiFormatDataSchemaResolver resolver = MultiFormatDataSchemaResolver.withBuiltinFormats(_resolverPath);
-      PegasusSchemaParser parser = AbstractSchemaParser.parserForFileExtension(_sourceFormat, resolver);
-      parser.parse(new FileInputStream(sourceFile));
-      LOGGER.debug("Loaded source schema: {}, from location: {}", schemaFullname, sourceFile.getAbsolutePath());
-      NamedDataSchema schema = checkForErrorsAndGetTopLevelSchema(sourceFile, schemaFullname, parser, errorMessages);
-      if (schema != null)
+      DataSchema schema = entry.getKey();
+      DataSchemaLocation location = entry.getValue();
+      if (!location.getSourceFile().getAbsolutePath().endsWith(_sourceFormat))
       {
-        topLevelTranslatedSchemas.put(schemaFullname, new SchemaInfo(schema, sourceFile, encode(schema)));
+        // Skip source files not matching the format specified.
+        continue;
+      }
+      if (schema instanceof NamedDataSchema)
```

Comment:
Parsing schemas in this way, we may not get expected result. 
DataSchemaParser.ParseResult (parsedSources) also contains embedded name schemas which are unnecessary to be top level schemas.

Reply:
+1, please consider the side effect of generated non-top level files

Reply:
ParseResult will not contain nested schemas. See relevant logic here:
https://github.com/linkedin/rest.li/blob/74121d54a156824aa2c7db15ade34e8852b1ea08/generator/src/main/java/com/linkedin/pegasus/generator/FileFormatDataSchemaParser.java#L259

It only adds top-level schemas to the parse result.

Reply:
Look at here: https://github.com/linkedin/rest.li/blob/894952afb12760a6d3cace8ff5a2ae44ed2149d4/generator/src/main/java/com/linkedin/pegasus/generator/FileFormatDataSchemaParser.java#L139.

Reply:
I think ybi means in the end of the parsing process, there will still be more non-top level schemas

Reply:
Good catch. Infact the schema translator tests were failing because of that!
Fixed the logic now to check for top level schemas and also ignore schemas loaded from resolver path (not in source dir).

Reply:
In the isTopLevelSchema method, we have already checked whether schema is NamedDataSchema, do we still need to do the same check here?

Reply:
removed the check, it was redundant.